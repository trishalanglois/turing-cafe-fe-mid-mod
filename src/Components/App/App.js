import React, { Component } from 'react';
import './App.css';
import Form from '../Form/Form'
import ReservationsContainer from '../ReservationsContainer/ReservationsContainer'


class App extends Component {
  constructor() {
    super();
    this.state = {
      allReservations: []
    }
  }

    componentDidMount() {
      fetch('http://localhost:3001/api/v1/reservations')
        .then(response => response.json())
        .then(data =>
          this.setState({
            allReservations: data
          })
        )

    }

    getUpdatedData() {
      fetch('http://localhost:3001/api/v1/reservations')
        .then(response => response.json())
        .then(data =>
          this.setState({
            allReservations: data
          })
        )
    }

    submitReservation = (reservation) => {
      this.setState({
        allReservations: [...this.state.allReservations, reservation]
      })
      fetch('http://localhost:3001/api/v1/reservations', {
      	method: 'POST',
      	headers: {
      		'Content-Type': 'application/json'
      	},
      	body: JSON.stringify({...reservation, id: Date.now()}),
      })
      	.then(response => response.json())
        .then(data => console.log(data))
      	.catch(error => console.log(error))
      this.getUpdatedData();
    }

    removeReservation = (event) => {
      console.log('cancel button has been clicked');
    }

    render() {
      return (
        <div className="App">
          <h1 className='app-title'>Turing Cafe Reservations</h1>
          <div className='resy-form'>
            <Form submitReservation={this.submitReservation}/>
          </div>

          <div className='resy-container'>
          <ReservationsContainer
            reservations={this.state.allReservations}/>
          </div>

        </div>
  )}
}

export default App;
