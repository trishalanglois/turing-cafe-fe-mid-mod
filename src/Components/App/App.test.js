import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';


describe('App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match the snapshot', () => {
    const wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot();
  })


//NEEDS ASYNC TESTING FOR FETCH
  // it('should update state when getUpdatedData is called', () => {
  //   const wrapper = shallow(<App />);
  //   const mockOGState = [];
  //   const mockNewState = [{
  //       id: 1,
  //       name: "Christie",
  //       date: "12/29",
  //       time: "7:00",
  //       number: 12
  //   }]
  //
  //   wrapper.setState({reservations: mockOGState});
  //   expect(wrapper.state('reservations')).toEqual(mockOGState);
  //
  //   wrapper.instance().getUpdatedData(mockNewState);
  //   expect(wrapper.state('reservations')).toEqual(mockNewState)
  // })

})
