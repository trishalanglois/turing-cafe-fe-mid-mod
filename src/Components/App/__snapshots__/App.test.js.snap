// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<Fragment>
  <Form
    submitReservation={[Function]}
  />
  <ReservationsContainer
    reservations={Array []}
  />
</Fragment>
`;
